patches/poco.patch

```diff
@@ -166,11 +166,11 @@
 +POCO_TARGET_OSARCH  = $(ANDROID_ABI)
 +
 +ifeq ($(ANDROID_ABI),armeabi)
```

Comment:
Good observation. We haven't talked about this, but the idea would be to support `armeabi-v7a` and `arm64-v8a` right?

Reply:
Not necessarily for us - we only need `arm64-v8a`. But if it is easy to keep the build scripts compatible and support both (in terms of Android ABI, not in terms of the NDK version), I would prefer to do so in order to increase the number of potential users. Otherwise you would not have to add an `abi` option to [config.sh](https://github.com/Intermodalics/ros_android/blob/kinetic/config.sh).

Switching to Clang as a compiler is fine because gcc is deprecated in newer NDK versions or even has been removed.

Reply:
Now I see it; the problem is that `arch` flag is hardcoded to `arm64` in `build_library_with_toolchain`, so yes, if we want to support both of them we also need to pass `ANDROID_ABI` as a parameter.
Note that `armeabi` was removed after `ndk r17`, so we can omit that one and keep `arm64-v8a` and `armeabi-v7a`.